File: Backend/physics_constants.py
"""
Shared Physics Constants for SmartWatt Hybrid 0
This file is the "Ground Truth".
When the AI feels unsure (or for simple devices like Irons), 
it checks this file to know the correct Wattage.
"""

PHYSICS_DEFAULTS = {
    # MAJOR APPLIANCES
    'ac': {
        'reference_watts': 1000, # Per Ton (e.g. 1.5 Ton = 1500W base)
        'default_hours': 8
    },
    'fridge': {
        'reference_watts': 40,   # Baseline active draw (approx for 250L) - scaling handled by logic
        'default_hours': 24
    },
    'washing_machine': {
        'reference_watts': 500,  # Average cycle power
        'default_hours': 1.0     # Per day approx (derived from cycles)
    },
    'water_heater': {
        'reference_watts': 2000,
        'default_hours': 1.5
    },
    'water_pump': {
        'reference_watts': 746, # 1 HP
        'default_hours': 0.5
    },
    'television': {
        'reference_watts': 100, # Approx for 43-50 inch LED
        'default_hours': 4
    },

    # LIGHTING & FANS
    'ceiling_fan': {
        'reference_watts': 75,
        'default_hours': 12
    },
    'led_light': {
        'reference_watts': 10,
        'default_hours': 6
    },
    'tube_light': {
        'reference_watts': 40,
        'default_hours': 6
    },
    'cfl_bulb': {
        'reference_watts': 20,
        'default_hours': 6
    },

    # KITCHEN
    'microwave': {
        'reference_watts': 1200,
        'default_hours': 0.3
    },
    'mixer_grinder': {
        'reference_watts': 750,
        'default_hours': 0.5
    },
    'kettle': {
        'reference_watts': 1500,
        'default_hours': 0.2
    },
    'induction': {
        'reference_watts': 1500, # Variable 1200-2000
        'default_hours': 1.0
    },
    'rice_cooker': {
        'reference_watts': 700,
        'default_hours': 0.5
    },

    # OTHER
    'desktop': {
        'reference_watts': 200,
        'default_hours': 4
    },
    'laptop': {
        'reference_watts': 50,
        'default_hours': 4
    },
    'iron': {
        'reference_watts': 1000,
        'default_hours': 0.2
    }
}

File: Frontend/src/lib/physicsConstants.ts
/** 
 * Centralized Physics Constants for SmartWatt
 * ===========================================
 * Defines baseline wattage and efficiency factors for all appliances.
 * Source: BEE India Standards & Default Market Averages (2025)
 */

export const PHYSICS_DEFAULTS: Record<string, number> = {
    ceiling_fan: 75,      // Standard inductive fan
    led_light: 10,
    cfl_bulb: 15,
    tube_light: 40,
    ac: 1500,             // 1.5 Ton Non-Inverter Baseline
    refrigerator: 200,    // 250L Frost Free Baseline
    washing_machine: 500, // Top Load Washing Cycle
    television: 100,      // 43" LED Baseline
    desktop: 200,
    laptop: 50,
    water_heater: 2000,   // 25L Storage Geyser
    water_pump: 750,      // 1 HP Pump
    iron: 1000,
    chiller: 1500,
    induction: 1500,
    mixer_grinder: 750,
    microwave: 1200,
    kettle: 1500,
    rice_cooker: 700,
    toaster: 800,
    food_processor: 500,
    hair_dryer: 1200,
    vacuum: 1000
};

// Special Constants used in logic
export const PHYSICS_RATIOS = {
    BLDC_FAN_WATTS: 28,
    INSTANT_GEYSER_WATTS: 3000,
    TV_WATTS_PER_INCH: 2.0,
    FRIDGE_BASELINE_CAPACITY: 250,
    FRIDGE_HOURLY_BASE: 40, // Approx average hourly draw
    PUMP_HP_TO_WATTS: 746,
    AC_TON_TO_WATTS: 1000   // Rough baseline for high efficiency
};
